image/default_image.go

```diff
@@ -169,66 +169,73 @@ func (d DefaultImageService) downloadLayers(named reference.Named, manifest sche
 	errorCh := make(chan error, 2*len(manifest.Layers))
 	defer func() {
 		close(errorCh)
-		lerr := errors.New("failed to upload layers")
+		lerr := errors.New("failed to download layers")
 		for e := range errorCh {
 			err = errors.Wrap(e, lerr.Error())
 			lerr = err
 		}
 	}()
 
-	for _, layer := range manifest.Layers {
-		hex := layer.Digest.Hex()
-		shortHex := hex
-		if len(shortHex) > 12 {
-			shortHex = shortHex[0:12]
-		}
-
-		roLayer := layerStore.Get(store.LayerID(layer.Digest))
-		if roLayer != nil {
-			flow.ShowMessage(shortHex+" already exists", nil)
-			continue
-		}
-
-		// get layers stream first
-		blobReader, err := d.registry.DownloadLayer(context.Background(), named.Repo(), layer.Digest)
-		if err != nil {
-			flow.ShowMessage(shortHex+fmt.Sprintf(" failed to pull layer, err: %s", err), nil)
-			errorCh <- err
-			continue
-		}
+	for _, lyr := range manifest.Layers {
+		func(layer distribution.Descriptor) {
```

Comment:
why you need this closure?

Reply:
for making layer a local valiable, cuz Action will be executing in goroutines